### PR TITLE
[Ready] All improvised guns are now slap crafted, as well as adding in makeshift pistols

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
## About The Pull Request

No longer can you just "craft" the improvised shotgun and rifle you can slap each part into one a other and make your own fire arm! Just like back in the day you now have to craft it using what is needed rather then waiting 100 desceconds for a timer to fill.
Added in a makeshift pistol that has 9mm 10mm and .45, one shot at a time
Added a makeshift high caliber pistol that holds .38 .357 and n762, one shot at a time
The makeshift pistols have fail odds, failing does a simular action to what detectives does

## Why It's Good For The Game

Adds more skill into crafting these weapons, as well as more guns to craft to stick it to sec when you viva!
The improvised pistols are balanced rather well imo and moving the improvised rifle and shotgun out of crafting makes them more of a hidden craftable, for those that know the way of slap crafting.

## Changelog
:cl:
add: High caliber and normal makeshift pistols - only gettable via admins
balance: Improvised shotgun and rifle are now slap craftable 
/:cl:

### CODE REVIEW

**Comment by Trilbyspaceclone** [comment]
Its all 100% tested! (Unless I somehow missed something)

**Review by silicons — CHANGES_REQUESTED** [review]
No makeshift pistols/revolvers.

Leave it adminspawn only/admingive only for crafting recipe if you want, I do not want this roundstart.

File: code/datums/components/crafting/guncrafting.dm [hunk 1]
```diff
@@ -2,14 +2,117 @@
...
+/obj/item/weaponcrafting/receiver/stock/barrled/gun/long
+	name = "rifle gun assembly"
+	desc = "A classic rifle stock that doubles as a grip, roughly carved out of wood, this one has a modular receiver firmly attached with an exstended barrel."
+	icon_state = "mounted_receiver_piped_wrapped"
+
+/obj/item/weaponcrafting/receiver/stock/barrled/gun/long/attackby(obj/item/D, mob/user, params)
+	if(istype(D, /obj/item/screwdriver))
+		to_chat(user, "<span class='notice'>You start attaching the gun to the frame...</span>")
+		if(D.use_tool(src, user, 40, volume=100))
+			to_chat(user, "<span class='notice'>Improvised Rifle Completed!</span>")
```
Comment (silicons) [inline code/datums/components/crafting/guncrafting.dm]: Don't capitalize every word.

File: code/datums/components/crafting/guncrafting.dm [hunk 2]
```diff
@@ -2,14 +2,117 @@
...
+/obj/item/weaponcrafting/receiver/stock/barrled/gun/short
+	name = "small gun assembly"
+	desc = "A classic rifle stock that doubles as a grip, roughly carved out of wood, this one has a modular receiver firmly attached with a shortend barrel. Using a drill will allow the gun to use high caliber rounds well a screwdriver will be low caliber."
+	icon_state = "pistol"
+
+/obj/item/weaponcrafting/receiver/stock/barrled/gun/short/attackby(obj/item/D, mob/user, params)
+	if(istype(D, /obj/item/screwdriver))
+		to_chat(user, "<span class='notice'>You start attaching the gun to the frame...</span>")
+		if(D.use_tool(src, user, 40, volume=100))
+			to_chat(user, "<span class='notice'>Improvised Pistol Completed!</span>")
```
Comment (silicons) [inline code/datums/components/crafting/guncrafting.dm]: Ditto

File: code/datums/components/crafting/guncrafting.dm [hunk 3]
```diff
@@ -2,14 +2,117 @@
...
+/obj/item/weaponcrafting/receiver/stock/barrled/gun/long/attackby(obj/item/D, mob/user, params)
+	if(istype(D, /obj/item/screwdriver))
+		to_chat(user, "<span class='notice'>You start attaching the gun to the frame...</span>")
+		if(D.use_tool(src, user, 40, volume=100))
+			to_chat(user, "<span class='notice'>Improvised Rifle Completed!</span>")
+			var/obj/item/gun/ballistic/shotgun/boltaction/improvised/B = /obj/item/gun/ballistic/shotgun/boltaction/improvised
+			new B(user.loc, 1)
+			qdel(src)
+
+/obj/item/weaponcrafting/receiver/stock/barrled/gun/short
```
Comment (silicons) [inline code/datums/components/crafting/guncrafting.dm]: Misspelled typepath

File: code/datums/components/crafting/guncrafting.dm [hunk 4]
```diff
@@ -2,14 +2,117 @@
...
+		to_chat(user, "<span class='notice'>You start attaching the gun to the frame...</span>")
+		if(D.use_tool(src, user, 40, volume=100))
+			to_chat(user, "<span class='notice'>Improvised Pistol Completed!</span>")
+			var/obj/item/gun/ballistic/revolver/makeshift_pistol/M = /obj/item/gun/ballistic/revolver/makeshift_pistol
+			new M(user.loc, 1)
+			qdel(src)
+	if(istype(D, /obj/item/surgicaldrill))
+		to_chat(user, "<span class='notice'>You start attaching the gun to the frame...</span>")
+		if(D.use_tool(src, user, 40, volume=100))
+			to_chat(user, "<span class='notice'>Improvised High Caliber Pistol Completed!</span>")
```
Comment (silicons) [inline code/datums/components/crafting/guncrafting.dm]: Ditto

File: code/datums/components/crafting/guncrafting.dm [hunk 5]
```diff
@@ -2,14 +2,117 @@
...
+			new I(user.loc, 1)
+			qdel(src)
+	if(istype(D, /obj/item/circular_saw))
+		if(D.use_tool(src, user, 40, volume=100))
+			to_chat(user, "<span class='notice'>The barrle is sawn off making a smaller gun!</span>")
+			var/obj/item/weaponcrafting/receiver/stock/barrled/gun/short/L = /obj/item/weaponcrafting/receiver/stock/barrled/gun/short
+			new L(user.loc, 1)
+			qdel(src)
+	if(istype(D, /obj/item/pipe))
+		to_chat(user, "<span class='notice'>The barrle is exstended making a longer gun!</span>")
```
Comment (silicons) [inline code/datums/components/crafting/guncrafting.dm]: Typo

File: code/datums/components/crafting/guncrafting.dm [hunk 6]
```diff
@@ -2,14 +2,117 @@
...
+	if(istype(D, /obj/item/screwdriver))
+		to_chat(user, "<span class='notice'>You start attaching the gun to the frame...</span>")
+		if(D.use_tool(src, user, 40, volume=100))
+			to_chat(user, "<span class='notice'>Improvised Shotgun Completed!</span>")
+			var/obj/item/gun/ballistic/revolver/doublebarrel/improvised/I = /obj/item/gun/ballistic/revolver/doublebarrel/improvised
+			new I(user.loc, 1)
+			qdel(src)
+	if(istype(D, /obj/item/circular_saw))
+		if(D.use_tool(src, user, 40, volume=100))
+			to_chat(user, "<span class='notice'>The barrle is sawn off making a smaller gun!</span>")
```
Comment (silicons) [inline code/datums/components/crafting/guncrafting.dm]: Typo

File: code/datums/components/crafting/guncrafting.dm [hunk 7]
```diff
@@ -2,14 +2,117 @@
...
+/obj/item/weaponcrafting/receiver/stock/barrled/attackby(obj/item/W, mob/user, params)
+	..()
+	if(istype(W, /obj/item/stack/packageWrap))
+		var/obj/item/weaponcrafting/receiver/stock/barrled/gun/G = /obj/item/weaponcrafting/receiver/stock/barrled/gun
+		if(W.use(15))
+			to_chat(user, "<span class='notice'>You add [W] to [src]!</span>")
+			qdel(src)
+			new G(user.loc, 1)
+		else
+			to_chat(user, "<span class='warning'>You need at least ten lengths of package wrap if you want to make a attach this pipe!</span>")
```
Comment (silicons) [inline code/datums/components/crafting/guncrafting.dm]: Grammar

**Comment by IronEleven** [comment]
How does this make pipe guns any harder to make aside from a few days of confused mhelps?

**Comment by Trilbyspaceclone** [comment]
@IronEleven its faster, and thats it.

**Review by silicons — CHANGES_REQUESTED** [review]
there's still typos everywhere, there's still free for all improvised pistols/revolver recipes

i already said no more improvised guns it's just unnecessary the greytide has enough weapons. why the hell do we need more?

File: code/datums/components/crafting/recipes/recipes_weapon_and_ammo.dm [hunk 8]
```diff
@@ -244,6 +244,36 @@
...
+				/obj/item/pen = 1,
+				/obj/item/stack/packageWrap = 25)
+	tools = list(TOOL_SCREWDRIVER)
+	time = 300
+	category = CAT_WEAPONRY
+	subcategory = CAT_WEAPON
+
+/datum/crafting_recipe/ispistol
+	name = "Improvised Pistol"
+	always_availible = FALSE
```
Comment (Trilbyspaceclone) [inline code/datums/components/crafting/recipes/recipes_weapon_and_ammo.dm]: @kevinz000 What...

**Comment by Ghommie** [comment]
Reopening in faith that this whole crafting code will be improved into a datum or something more acceptable.

File: code/datums/components/crafting/guncrafting.dm [hunk 9]
```diff
@@ -2,20 +2,130 @@
...
-	desc = "A long pice of silk looks like cable coil."
-	icon = 'icons/obj/improvised.dmi'
-	icon_state = "silkstring"
\ No newline at end of file
+/obj/item/weaponcrafting/stock/attackby(obj/item/S, mob/user, params)
+	..()
+	if(istype(S, /obj/item/pipe))
+		return
+	if(istype(S, /obj/item/weaponcrafting/receiver))
+		var/obj/item/weaponcrafting/receiver/stock/R = /obj/item/weaponcrafting/receiver/stock
```
Comment (Ghommie) [inline code/datums/components/crafting/guncrafting.dm]: Just remove this line use the Path in place of R on the `new` procedure.

This also applies to most, if not all, other steps.

File: code/datums/components/crafting/guncrafting.dm [hunk 10]
```diff
@@ -2,20 +2,130 @@
...
 	icon_state = "riflestock"
 
-/obj/item/weaponcrafting/silkstring
-	name = "silkstring"
-	desc = "A long pice of silk looks like cable coil."
-	icon = 'icons/obj/improvised.dmi'
-	icon_state = "silkstring"
\ No newline at end of file
+/obj/item/weaponcrafting/stock/attackby(obj/item/S, mob/user, params)
+	..()
```
Comment (Ghommie) [inline code/datums/components/crafting/guncrafting.dm]: Always set the parent return call as the standard return value unless you have a reason not to.
`. = ..()`, not `..()`.
This also applies to other steps.

File: code/datums/components/crafting/guncrafting.dm [hunk 11]
```diff
@@ -2,20 +2,130 @@
...
+		else
+			to_chat(user, "<span class='warning'>You need at least ten lengths of package wrap if you want to attach this pipe!</span>")
+
+/obj/item/weaponcrafting/receiver/stock/barreled/gun
+	name = "gun assembly"
+	desc = "A classic rifle stock that doubles as a grip, roughly carved out of wood. This one has a modular receiver firmly attached. Adding a new pipe would allow for a rifle to be made."
+	icon_state = "mounted_receiver_piped_wrapped"
+
+/obj/item/weaponcrafting/receiver/stock/barreled/gun/attackby(obj/item/D, mob/user, params)
+	if(istype(D, /obj/item/screwdriver))
```
Comment (Ghommie) [inline code/datums/components/crafting/guncrafting.dm]: Use tool behaviours checks or the appropriate tools act procedure if possible, not type checks.
This also applies to all other steps.

File: code/datums/components/crafting/recipes/recipes_weapon_and_ammo.dm [hunk 12]
```diff
@@ -260,6 +260,36 @@
...
+	reqs = list(/obj/item/weaponcrafting/receiver = 1,
+				/obj/item/pipe = 1,
+				/obj/item/weaponcrafting/stock = 1,
+				/obj/item/stack/rods = 1,
+				/obj/item/pen = 1,
+				/obj/item/stack/packageWrap = 25)
+	tools = list(TOOL_SCREWDRIVER)
+	time = 300
+	category = CAT_WEAPONRY
+	subcategory = CAT_WEAPON
```
Comment (Ghommie) [inline code/datums/components/crafting/recipes/recipes_weapon_and_ammo.dm]: Remove, these shouldn't be craftable this way.

File: code/datums/components/crafting/recipes/recipes_weapon_and_ammo.dm [hunk 13]
```diff
@@ -260,6 +260,36 @@
...
+
+/datum/crafting_recipe/ispistol
```
Comment (Ghommie) [inline code/datums/components/crafting/recipes/recipes_weapon_and_ammo.dm]: Same here.

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Comment by Trilbyspaceclone** [comment]
@Ghommie What other changes are needed?

**Comment by BobJoga** [comment]
I don't see how this makes crafting any harder. It justs makes coding and spriting harder and more painful.
-1, pretty bad

**Pull request closed without merging**